controllers/profilecatalogsource_controller.go

```diff
@@ -63,6 +77,28 @@ func (r *ProfileCatalogSourceReconciler) Reconcile(ctx context.Context, req ctrl
 	logger.Info("updating catalog entries", "profiles", pCatalog.Spec.Profiles)
 	r.Profiles.Update(pCatalog.Name, pCatalog.Spec.Profiles...)
 
+	timeout := time.Minute * 2
+	interval := time.Second * 5
+	gitRepoManager := gitrepository.NewManager(ctx, pCatalog.Namespace, r.Client, timeout, interval)
+	scanner := scanner.New(gitRepoManager, &git.Client{}, http.DefaultClient, logger)
+	for _, repo := range pCatalog.Spec.Repos {
+		logger.Info("scan repo for profiles", "repo", repo)
+		var secret *corev1.Secret
+		if repo.SecretRef != nil {
+			secret = &corev1.Secret{}
+			err := r.Client.Get(ctx, client.ObjectKey{Name: repo.SecretRef.Name, Namespace: req.Namespace}, secret)
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to find secret for repo %v: %w", repo, err)
+			}
+		}
+
+		profiles, err := scanner.ScanRepository(repo, secret)
+		if err != nil {
+			return ctrl.Result{}, err
```

Comment:
Please add some logging here, so we know that's happening.

Reply:
returning an err does produce a log, and we log above whats happening, so I think its okay?

Reply:
oh okay. :)